### ISSUE TITLE
[23330] On hover tooltip for contact permissions

### ISSUE THREAD

**Issue by Aycorn** [DeskDirector/UserVoice#475]
Feature request to have the current "Is overriding company permission settings" on hover text that displays for any enabled/green icons display as a list of the enabled override features:

![image](https://user-images.githubusercontent.com/30588784/91228007-9ed56400-e77b-11ea-8601-ebd8ae6221bf.png)

Would display a short list of enabled features:

![image](https://user-images.githubusercontent.com/30588784/91228067-b57bbb00-e77b-11ea-8acb-8f2c9423004b.png)

**Comment by Nness** [comment]
This is tough one, design has been changed several times already. I doubt tooltip can help. How many will hover and read the tooltip?

**Comment by cdandrea** [comment]
I suspect the use case will be limited to those looking to make bulk changes across tens or hundreds of contacts. **Does the effort to implement justify the return? Perhaps not.**

Part of the motivation is that when clicking into a contact, backing out resets the contact view, which is pretty frustrating. Yes, using your browser's back button will maintain your previous search, but (expectedly) clicking on the "Contacts" breadcrumb reloads. If the contact list retained its previous state (even when clicking on the Contacts link along top of the contact), it would go probably eliminate this need.

What prompted the request was trying to track permission config overwrites against a separate export of ConnectWise contacts that had their contact types set to identify their intended overwrite options. This would normally be a maintenance task going forward, but establishing the initial overwrite options was a bulk exercise, across 500+ contacts. Not fun.

I opted to export the contact list from CW, which includes the contact record ID, and also exported the DD contacts permissions table. I realized that I could check off all of the overwrite permissions, which would include **any** contacts that had **any** overwrites. This CSV also included the CW contact record ID so I could VLOOKUP and highlight any DD contacts that had not yet been set.

In other words, I've figured out a way around this, to make the verification process manageable. Otherwise, drilling into each contact was a rather painful process, hence the ask to have that quick spot check via tooltip to save the clicking.

In summary, I won't be offended if you close this request ;)

**Comment by Nness** [comment]
There are two topic here.

### Remain search state
This is done by using browser storage in TECH, while using browser history for admin portal. Personally, I prefer browser history rather than browser storage. When click on *contacts`, it should be on a clean state. Which is same to many other modern application.

For TECH, as I have picked up that project, in the future I would like to adjust it to use history rather than storage. But that's only until we switch the framework we are using. So its still far in the future.

### Bulk editing
As designer for admin portal, from beginning, companies, contacts been planned to have bulk editing. Contact list has multi select, it is intended for such feature.

The priority is medium to low, as we assume most of customer won't need modify it constantly. Only during initial on boarding. Of course this may not be the case for everyone.

I'm currently working on member list in admin portal, which I have several other task need to be accomplish. I cannot guarantee you on estimate, this is always on my personal table to get it done as soon as an gap is open.

Planned bulk editing is for any config that can be override to be modified in bulk. Either unset in bulk or set in bulk.

**Comment by kcjones76** [comment]
I honestly appreciate you entertaining my requests and providing more than one-sentence answers. I provides a good picture into where the devs is at. I actually agree with the use of the browser history. There’s many that browser-based that break, e.g. CW is a prime is a prime example. Much quicker to use your shortcuts or mouse buttons versus having to click around on the pages.




[cid:logo_mbu_oet_codetwo_415x75_c25eb2e2-927f-4ecc-a9b4-08fb1d46b845.png]<https://mbu.ca>

Casey D'Andrea | Infrastructure Lead
T: 416.245.0709 x4256 | F: 416.849.0145 | E: cdandrea@mbu.ca


[cid:icon_facebook_32x32_b999bfb7-1286-410c-b529-f34697ed15bd.png]<https://www.facebook.com/myblueumbrella>   [cid:icon_twitter_32x32_859463eb-74e2-4ab2-9538-3d780c202a85.png] <https://twitter.com/myblueumbrella>    [cid:icon_linkedin_32x32_985edeb9-d72d-453e-9b02-f3d9949c3d1c.png] <https://www.linkedin.com/company/my-blue-umbrella>       [cid:logo_doamazingthings_happyyo_tagline_32x190_d1edbcf5-c4a2-4746-ae67-fd9f29c9262d.png] <http://www.doamazingthings.ca/>



________________________________
